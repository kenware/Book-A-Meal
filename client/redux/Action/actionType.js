
export const LOAD_MEALS = 'LOAD_MEALS';
