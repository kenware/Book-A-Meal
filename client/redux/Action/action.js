
import * as types from './actionType';
import auth from '../../authenticate/auth';
import history from '../../history';

export const loadMostOrdered = mostOrder => ({ type: types.LOAD_MOST_ORDERED, mostOrder });
export const loadErrorMessage = errorMessage => ({ type: types.LOAD_ERROR_MESSAGE, errorMessage });
export const loadSuccessMessage = successMessage => ({
  type: types.LOAD_SUCCESS_MESSAGE, successMessage
});
export const loadAllMeals = meals => ({ type: types.LOAD_ALL_MEALS, meals });
export const loadNotifications = notifics => ({ type: types.LOAD_NOTIFICATIONS, notifics });
export const loadUser = user => ({ type: types.LOAD_USER, user });

export const loadMostOrderedMeal = () => (dispatch) => {
  fetch('/api/v1/mostOrder/meals/6')
    .then(res => res.json())
    .then(mostOrder =>
      dispatch(loadMostOrdered(mostOrder)));
};

export const register = (name, username, email, password) => (dispatch) => {
  fetch('/api/v1/auth/signup', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      email,
      password,
      name,
      username
    })
  })
    .then(res => res.json())
    .then((res) => {
      if (res.message) {
        return dispatch(loadErrorMessage({ registerError: res.message }));
      }
      auth.setAuth(res.token, res.username, res.id, res.role, res.image);
      if (res.role === 'admin') {
        return history.push('/admin');
      }
      return history.push('/dashboard');
    });
};

export const login = (username, password) => (dispatch) => {
  fetch('/api/v1/auth/signin', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      password,
      username
    })
  })
    .then(res => res.json())
    .then((response) => {
      if (!response.username) {
        return dispatch(loadErrorMessage({ loginError: response.message }));
      }
      auth.setAuth(response.token, response.username, response.id, response.role, response.image);
      if (response.role === 'admin') {
        return history.push('/admin');
      }
      history.push('/dashboard');
    });
};
export const getNotifications = () => (dispatch) => {
  fetch('/api/v1/notifications', {
    headers: {
      authorization: auth.getToken()
    }
  })
    .then(res => res.json())
    .then(notifics => dispatch(loadNotifications(notifics)));
};
export const refreshToken = role => (dispatch) => {
  fetch('/api/v1/refresh', {
    headers: {
      authorization: auth.getToken()
    }
  })
    .then(res => res.json())
    .then((newUser) => {
      if (newUser.message) {
        dispatch(loadErrorMessage({ authError: 'Your Session Expired, Please Login' }));
        auth.logOut();
        return history.push('/login');
      }
      if (newUser.role === 'user' && newUser.role !== role) {
        auth.setRefresh(newUser.token, newUser.image);
        return history.push('/dashboard');
      }
      auth.setRefresh(newUser.token, newUser.image);
    });
};
export const upgrade = () => (dispatch) => {
  fetch('/api/v1/auth/admin', {
    method: 'POST',
    headers: {
      authorization: auth.getToken()
    }
  })
    .then(res => res.json())
    .then((res) => {
      if (!res.setAdmin) {
        return dispatch(loadErrorMessage({ upgradeError: res.message }));
      }
      auth.logOut();
      const {
        username, id, role, image
      } = res.setAdmin;
      auth.setAuth(res.token, username, id, role, image);
      dispatch(loadSuccessMessage({ upgradeSuccess: 'Upgrade successfull' }));
    });
};

export const clearMessages = () => (dispatch) => {
  dispatch(loadErrorMessage([]));
  dispatch(loadSuccessMessage([]));
};

export const getUser = () => (dispatch) => {
  fetch('/api/v1/auth/user', {
    headers: {
      authorization: auth.getToken()
    }
  })
    .then(res => res.json())
    .then(user => dispatch(loadUser(user)));
};

export const updateProfile = payload => (dispatch) => {
  fetch('/api/v1/auth/update', {
    headers: {
      authorization: auth.getToken()
    },
    method: 'POST',
    body: payload
  })
    .then(res => res.json())
    .then((user) => {
      if (user.message) {
        return dispatch(loadErrorMessage({ updateError: user.message }));
      }
      if (user.image) {
        window.localStorage.setItem('image', user.image);
      }
      getUser();
      return dispatch(loadSuccessMessage({ updateSuccess: 'User updated!' }));
    });
};
export const resetLink = emailOrUsername => (dispatch) => {
  fetch('/api/v1/auth/resetLink', {
    headers: {
      authorization: auth.getToken(),
      'Content-Type': 'application/json'
    },
    method: 'POST',
    body: JSON.stringify({
      emailOrUsername
    })
  })
    .then(res => res.json())
    .then((link) => {
      if (link.message) {
        return dispatch(loadErrorMessage({ resetError: link.message }));
      }
      return dispatch(loadSuccessMessage({ resetSuccess: 'Reset Link sent to your email' }));
    });
};
