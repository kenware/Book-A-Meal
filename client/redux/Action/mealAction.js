
import * as types from './actionType';
import auth from '../../authenticate/auth';

export const loadMostOrdered = mostOrder => ({ type: types.LOAD_MOST_ORDERED, mostOrder });
export const loadErrorMessage = errorMessage => ({ type: types.LOAD_ERROR_MESSAGE, errorMessage });
export const loadSuccessMessage = successMessage => ({
  type: types.LOAD_SUCCESS_MESSAGE, successMessage
});
export const loadAllMeals = meals => ({ type: types.LOAD_ALL_MEALS, meals });

export const loadMostOrderedMeal = limit => dispatch => window.fetch(`/api/v1/mostOrder/meals?limit=${limit}`)
  .then(res => res.json())
  .then(mostOrder =>
    dispatch(loadMostOrdered(mostOrder)));

export const getAllMeals = (limit, offset) => dispatch => window.fetch(`/api/v1/meals?limit=${limit}&offset=${offset}`, {
  headers: {
    authorization: auth.getToken()
  }
})
  .then(res => res.json())
  .then((meals) => {
    if (!meals.message) {
      return dispatch(loadAllMeals(meals));
    }
    return dispatch(loadAllMeals({ rows: [], count: 0 }));
  });

export const createMeal = payload => dispatch => window.fetch('/api/v1/meals', {
  method: 'POST',
  headers: {
    authorization: auth.getToken()
  },
  body: payload
})
  .then(res => res.json())
  .then((response) => {
    if (response.message) {
      return dispatch(loadErrorMessage({ createMealError: response.message }));
    }
    dispatch(getAllMeals(6, 0));
    dispatch(loadSuccessMessage({ createMealSuccess: 'Meal Successfully Created' }));
  });

export const deleteMeal = id => dispatch => window.fetch(`/api/v1/meals/${id}`, {
  method: 'DELETE',
  headers: {
    authorization: auth.getToken()
  }
})
  .then(res => res.json())
  .then((meals) => {
    dispatch(loadSuccessMessage(meals));
    dispatch(getAllMeals(6, 0));
  });

export const updateMeal = (id, payload) => dispatch => window.fetch(`/api/v1/meals/${id}`, {
  method: 'PUT',
  headers: {
    authorization: auth.getToken()
  },
  body: payload
})
  .then(res => res.json())
  .then((response) => {
    if (response.message) {
      return dispatch(loadErrorMessage({ updateMealError: response.message }));
    }
    dispatch(getAllMeals(6, 0));
    return dispatch(loadSuccessMessage({ updateMealSuccess: 'Meal Successfully Updated' }));
  });
