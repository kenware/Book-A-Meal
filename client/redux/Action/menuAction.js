
import * as types from './actionType';
import auth from '../../authenticate/auth';

export const loadErrorMessage = errorMessage => ({ type: types.LOAD_ERROR_MESSAGE, errorMessage });

export const loadSuccessMessage = successMessage => ({
  type: types.LOAD_SUCCESS_MESSAGE, successMessage
});

export const loadTodayMenu = menu => ({ type: types.LOAD_TODAY_MENU, menu });

export const loadTodayMenuMeals = menuMeals => ({ type: types.LOAD_TODAY_MENUMEALS, menuMeals });
export const addMealToCart = cart => ({ type: types.ADD_MEAL_TO_CART, cart });

export const setMenu = (meals, title, orderBefore, sendEmail) => dispatch => window.fetch('/api/v1/menu', {
  method: 'POST',
  headers: {
    authorization: auth.getToken(),
    'Content-Type': 'application/json'
  },
  body: JSON.stringify({
    meals, title, orderBefore, sendEmail
  })
})
  .then(res => res.json())
  .then((response) => {
    if (!response.menu) {
      return dispatch(loadErrorMessage({ setMenuError: response.message }));
    }
    return dispatch(loadSuccessMessage({ setMenuSuccess: 'Meal added to menu' }));
  });

export const getMenu = (limit, offset) => dispatch => window.fetch(`/api/v1/menu?limit=${limit}&offset=${offset}`, {
  headers: {
    authorization: auth.getToken()
  }
})
  .then(res => res.json())
  .then((menu) => {
    if (menu.message) {
      return dispatch(loadErrorMessage({ getMenuError: 'Todays Menu is not set yet' }));
    }
    dispatch(loadTodayMenu(menu));
  });

export const getMenuMeals = (url, limit, offset) => dispatch => window.fetch(`${url}?limit=${limit}&offset=${offset}`, {
  headers: {
    authorization: auth.getToken()
  }
})
  .then(res => res.json())
  .then((menuMeals) => {
    if (menuMeals.message) {
      return dispatch(loadErrorMessage({ getMenuMealsError: 'Error fetching meals' }));
    }
    dispatch(loadTodayMenuMeals(menuMeals));
  });

export const clearMenuMeals = () => dispatch => dispatch(loadTodayMenuMeals({ meals: [] }));

export const addToCart = meals => dispatch => dispatch(addMealToCart(meals));
