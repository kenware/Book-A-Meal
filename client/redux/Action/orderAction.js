
import * as types from './actionType';
import auth from '../../authenticate/auth';

export const loadMostOrdered = mostOrder => ({ type: types.LOAD_MOST_ORDERED, mostOrder });
export const loadErrorMessage = errorMessage => ({ type: types.LOAD_ERROR_MESSAGE, errorMessage });
export const loadSuccessMessage = successMessage => ({
  type: types.LOAD_SUCCESS_MESSAGE, successMessage
});
export const loadTodayMenu = menu => ({ type: types.LOAD_TODAY_MENU, menu });
export const loadMyOrder = myOrder => ({ type: types.LOAD_MY_ORDER, myOrder });
export const loadAllOrder = allOrder => ({ type: types.LOAD_ALL_ORDER, allOrder });

export const orderMeal = (mealId, menuId, address, quantity) => (dispatch) => {
  fetch('/api/v1/orders', {
    method: 'POST',
    headers: {
      authorization: auth.getToken(),
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({
      mealId, menuId, address, quantity
    })
  })
    .then(res => res.json())
    .then((response) => {
      if (!response.status) {
        return dispatch(loadErrorMessage({ orderError: response.message }));
      }
      return dispatch(loadSuccessMessage({ orderSuccess: 'Order created' }));
      // return history.push('/dashboard');
    });
};
export const getMyOrder = () => (dispatch) => {
  fetch('/api/v1/user/orders', {
    headers: {
      authorization: auth.getToken()
    }
  })
    .then(res => res.json())
    .then((myOrder) => {
      if (myOrder.message) {
        dispatch(loadErrorMessage({
          myOrderError: 'OOps You Have Not Ordered A Meal'
        }));
        return dispatch(loadMyOrder([]));
      }
      dispatch(loadMyOrder(myOrder));
    });
};
export const getAllOrders = () => (dispatch) => {
  fetch('/api/v1/orders', {
    headers: {
      authorization: auth.getToken()
    }
  })
    .then(res => res.json())
    .then((allOrder) => {
      if (allOrder.message) {
        return dispatch(loadErrorMessage({
          allOrderError: 'OOps Users Have Not Ordered A Meal'
        }));
      }
      dispatch(loadAllOrder(allOrder));
    });
};
export const updateOrder = (id, quantity, address, status) => (dispatch) => {
  fetch(`/api/v1/orders/${id}`, {
    method: 'PUT',
    headers: {
      authorization: auth.getToken(),
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({
      id, status, address, quantity
    })
  })
    .then(res => res.json())
    .then((response) => {
      if (response.message) {
        return dispatch(loadErrorMessage({ updateError: response.message }));
      }
      dispatch(getMyOrder());
      dispatch(loadSuccessMessage({ updateSuccess: 'Order created' }));
    });
};
export const confirmStatus = id => (dispatch) => {
  fetch(`/api/v1/orderStatus/${id}`, {
    method: 'PUT',
    headers: {
      authorization: auth.getToken(),
      'Content-Type': 'application/json'
    }
  })
    .then(res => res.json())
    .then((response) => {
      if (response.message) {
        return dispatch(loadErrorMessage({ confirmError: response.message }));
      }
      dispatch(getMyOrder());
      dispatch(loadSuccessMessage({ confirmSuccess: 'Order created' }));
    });
};
