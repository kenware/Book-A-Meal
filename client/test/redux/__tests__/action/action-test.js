import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock from 'fetch-mock';
import * as actions from '../../../../redux/Action/action';
import * as types from '../../../../redux/Action/actionType';

// import expect from 'expect'; // You can use any testing library

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
// let auth = { getToken: jest.fn() };
describe('async actions test', () => {
  afterEach(() => {
    fetchMock.reset();
    fetchMock.restore();
  });
  it('should not create LOAD_NOTIFICATIONS when fetching notification with wrong token', () => {
    const notifics = { message: 'Please login!' };
    fetchMock
      .getOnce('/api/v1/notifications', { body: notifics });

    const expectedActions = [
      { type: types.LOAD_NOTIFICATIONS, notifics: [] }
    ];
    const store = mockStore({ notifics: [] });
    return store.dispatch(actions.getNotifications()).then(() => {
      // return of async actions
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  it('creates LOAD_NOTIFICATIONS when fetching notification has been done', () => {
    const notifics = { message: ['today menu is set'] };
    fetchMock
      .getOnce('/api/v1/notifications', { body: notifics });

    const expectedActions = [
      { type: types.LOAD_NOTIFICATIONS, notifics }
    ];
    const store = mockStore({ notifics: [] });
    return store.dispatch(actions.getNotifications()).then(() => {
      // return of async actions
      expect(store.getActions()).toEqual(expectedActions);
    });
  });
  it('creates LOAD_User when fetching user has been done', () => {
    const user = { message: ['do something'] };
    fetchMock
      .getOnce('/api/v1/auth/user', { body: user });

    const expectedActions = [
      { type: types.LOAD_USER, user }
    ];
    const store = mockStore({ mostOrder: [] });
    return store.dispatch(actions.getUser()).then(() => {
      // return of async actions
      expect(store.getActions()).toEqual(expectedActions);
    });
  });
  it('should fail to creates REGISTER_USER when fetching user has been done with invalid details', () => {
    const errorMessage = { message: 'error registering user' };
    fetchMock
      .post('/api/v1/auth/signup', { body: errorMessage });

    const expectedActions = [
      { type: types.LOAD_ERROR_MESSAGE, errorMessage: { registerError: 'error registering user' } }
    ];
    const store = mockStore([]);
    return store.dispatch(actions.register()).then(() => {
      // return of async actions
      expect(store.getActions()).toEqual(expectedActions);
    });
  });
  it('should REGISTER_USER when fetching user has been done', () => {
    global.history.pushState = jest.fn();
    const user = { name: 'ken', email: 'ken@gmail.com' };
    fetchMock
      .post('/api/v1/auth/signup', { body: user });

    const store = mockStore({ user: [] });
    return store.dispatch(actions.register()).then(() => {
      // return of async actions
      expect(store.getActions()).toMatchSnapshot();
    });
  });
  it('should REGISTER_USER when fetching user has been done and redirect to admin', () => {
    global.history.pushState = jest.fn();
    const user = { name: 'ken', email: 'ken@gmail.com', role: 'admin' };
    fetchMock
      .post('/api/v1/auth/signup', { body: user });

    const store = mockStore({ user: [] });
    return store.dispatch(actions.register()).then(() => {
      // return of async actions
      expect(store.getActions()).toMatchSnapshot();
    });
  });
  it('should LOGIN_USER when fetching user has been done and login normal user', () => {
    global.history.pushState = jest.fn();
    const user = { username: 'ken', password: 'ken@gmail.com' };
    fetchMock
      .post('/api/v1/auth/signin', { body: user });

    const store = mockStore({ user: [] });
    return store.dispatch(actions.login()).then(() => {
      // return of async actions
      expect(store.getActions()).toMatchSnapshot();
    });
  });
  it('should LOGIN_USER when fetching user has been done and login caterer', () => {
    global.history.pushState = jest.fn();
    const user = { username: 'ken', password: 'ken@gmail.com', role: 'admin' };
    fetchMock
      .post('/api/v1/auth/signin', { body: user });

    const store = mockStore({ user: [] });
    return store.dispatch(actions.login()).then(() => {
      // return of async actions
      expect(store.getActions()).toMatchSnapshot();
    });
  });
  it('should not LOGIN_USER with invalid detail', () => {
    const user = { message: 'Wrong details', role: 'admin' };
    fetchMock
      .post('/api/v1/auth/signin', { body: user });

    const expectedActions = [
      { type: types.LOAD_ERROR_MESSAGE, errorMessage: { loginError: 'Wrong details' } }
    ];

    const store = mockStore({ user: [] });
    return store.dispatch(actions.login()).then(() => {
      // return of async actions
      expect(store.getActions()).toEqual(expectedActions);
    });
  });
  it('should not refresh token if it is expired or do not exist', () => {
    const token = { token: '1234gdjjj4jjiu98u' };
    const data = { message: 'Your Session Expired' };
    fetchMock
      .get('/api/v1/refresh', { headers: { authorization: token }, body: data });

    const expectedActions = [
      { type: types.LOAD_ERROR_MESSAGE, errorMessage: { authError: 'Your Session Expired' } }
    ];

    const store = mockStore({ user: [] });
    return store.dispatch(actions.refreshToken()).then(() => {
      // return of async actions
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  it('should refresh token and redirect to dashbored for role user', () => {
    const token = { token: '1234gdjjj4jjiu98u' };
    const data = { role: 'user' };
    fetchMock
      .get('/api/v1/refresh', { headers: { authorization: token }, body: data });

    const store = mockStore({ user: [] });
    store.dispatch(actions.refreshToken('admin')).then(() => {
      // return of async actions
      expect(store.getActions()).toMatchSnapshot();
    });
    store.dispatch(actions.refreshToken('user')).then(() => {
      // return of async actions
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it('should not upgrade user that is not authenticated to admin', () => {
    const data = { message: 'Could not uprade to admin' };
    fetchMock
      .post('/api/v1/auth/admin', { body: data });

    const expectedActions = [
      { type: types.LOAD_ERROR_MESSAGE, errorMessage: { upgradeError: 'Could not uprade to admin' } }
    ];

    const store = mockStore({ upgrade: [] });
    return store.dispatch(actions.upgrade()).then(() => {
      // return of async actions
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  it('should upgrade normal user to admin', () => {
    const data = { setAdmin: { token: 'token', username: 'username' } };
    fetchMock
      .post('/api/v1/auth/admin', { body: data });

    const expectedActions = [
      { type: types.LOAD_SUCCESS_MESSAGE, successMessage: { upgradeSuccess: 'Upgrade successfull' } }
    ];

    const store = mockStore({ upgrade: [] });
    return store.dispatch(actions.upgrade()).then(() => {
      // return of async actions
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  it('should not update user profile with wrong credential and LOAD_ERROR_MESSAGE', () => {
    const data = { message: 'Could not update profile' };
    fetchMock
      .post('/api/v1/auth/update', { body: data });

    const expectedActions = [
      { type: types.LOAD_ERROR_MESSAGE, errorMessage: { updateError: 'Could not update profile' } }
    ];

    const store = mockStore({ update: [] });
    return store.dispatch(actions.updateProfile()).then(() => {
      // return of async actions
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  it('should update user profile and LOAD_SUCCESS_MESSAGE', () => {
    const data = { image: 'image_url' };
    fetchMock
      .post('/api/v1/auth/update', { body: data });

    const expectedActions = [
      { type: types.LOAD_SUCCESS_MESSAGE, successMessage: { updateSuccess: 'User updated!' } }
    ];

    const store = mockStore({ update: [] });
    return store.dispatch(actions.updateProfile()).then(() => {
      // return of async actions
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  it('should not send restlink on wrong user detail and LOAD_ERROR_MESSAGE', () => {
    const wrongData = { message: 'Wrong detail' };
    fetchMock
      .post('/api/v1/auth/resetLink', { body: wrongData });

    const expectedActions = [
      { type: types.LOAD_ERROR_MESSAGE, errorMessage: { resetError: 'Wrong detail' } }
    ];

    const store = mockStore({ link: [] });
    return store.dispatch(actions.resetLink()).then(() => {
      // return of async actions
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

  it('should send restlink to user', () => {
    const payload = { link: 'link_url' };
    fetchMock
      .post('/api/v1/auth/resetLink', { body: payload });
    const store = mockStore({ link: [] });
    return store.dispatch(actions.resetLink()).then(() => {
      // return of async actions
      expect(store.getActions()).toMatchSnapshot();
    });
  });
  it('should clear all messages', () => {
    const expectedActions = [
      { type: types.LOAD_ERROR_MESSAGE, errorMessage: {} }
    ];

    const store = mockStore({ message: [] });
    return store.dispatch(actions.clearMessages(), () => {
      // return of async actions
      expect(store.getActions()).toEqual(expectedActions);
    });
  });
});
