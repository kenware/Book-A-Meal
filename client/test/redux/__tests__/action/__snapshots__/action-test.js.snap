// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions test should dispatch LOGIN_USER action type on successful login 1`] = `Array []`;

exports[`async actions test should dispatch LOGIN_USER action type with caterer privilege 1`] = `Array []`;

exports[`async actions test should dispatch REGISTER_USER action type when fetching user has been done 1`] = `Array []`;

exports[`async actions test should dispatch REGISTER_USER action type with successful register with admin privilege 1`] = `Array []`;

exports[`async actions test should refresh token and redirect to dashbored for role user 1`] = `Array []`;

exports[`async actions test should refresh token and redirect to dashbored for role user 2`] = `Array []`;

exports[`async actions test should send restlink to user 1`] = `
Array [
  Object {
    "errorMessage": Object {
      "resetError": undefined,
    },
    "type": "LOAD_ERROR_MESSAGE",
  },
]
`;
